core/server/apps/amp/index.js

```diff
@@ -5,17 +5,20 @@ var router           = require('./lib/router'),
     config = require('../../config'),
     settingsCache = require('../../services/settings/cache');
 
+function ampRouter(req, res) {
+    if (settingsCache.get('amp') === true) {
+        return router.apply(this, arguments);
+    } else {
+        var redirectUrl = req.originalUrl.replace(/amp\/$/, '');
+        res.redirect(301, redirectUrl);
```

Comment:
Can you pls use `urlService.utils.redirect301`? This is a wrapper function, which allows us to define a behaviour for redirects.